packages/react-router/lib/hooks.tsx

```diff
@@ -401,11 +402,19 @@ export function useRoutes(
       matches.map((match) =>
         Object.assign({}, match, {
           params: Object.assign({}, parentParams, match.params),
-          pathname: joinPaths([parentPathnameBase, match.pathname]),
+          pathname: joinPaths([
+            parentPathnameBase,
+            // Re-encode pathnames that were decoded inside matchRoutes
+            navigator.encodeLocation(match.pathname).pathname,
+          ]),
           pathnameBase:
             match.pathnameBase === "/"
               ? parentPathnameBase
-              : joinPaths([parentPathnameBase, match.pathnameBase]),
+              : joinPaths([
+                  parentPathnameBase,
+                  // Re-encode pathnames that were decoded inside matchRoutes
+                  navigator.encodeLocation(match.pathnameBase).pathname,
+                ]),
```

Comment:
Need these to be re-encoded so that our `pathname.slice(parentPathnameBase.length)` call up on line 382 is correct.  It's slicing from an encoded `location.pathname` so needs to be using the length of an encoded `parentPathnameBase`